conda_lock/pypi_solver.py

```diff
@@ -0,0 +1,311 @@
+import re
+import sys
+
+from pathlib import Path
+from typing import Dict, List, Optional
+from urllib.parse import urldefrag
+
+from clikit.api.io.flags import VERY_VERBOSE
+from clikit.io import ConsoleIO, NullIO
+from packaging.tags import compatible_tags, cpython_tags
+from poetry.core.packages import Dependency, Package, ProjectPackage, URLDependency
+from poetry.installation.chooser import Chooser
+from poetry.installation.operations.uninstall import Uninstall
+from poetry.puzzle import Solver
+from poetry.repositories.pool import Pool
+from poetry.repositories.pypi_repository import PyPiRepository
+from poetry.repositories.repository import Repository
+from poetry.utils.env import Env
+
+from conda_lock import src_parser
+from conda_lock.src_parser.pyproject_toml import get_lookup as get_forward_lookup
+
+
+# NB: in principle these depend on the glibc in the conda env
+MANYLINUX_TAGS = ["1", "2010", "2014", "_2_17"]
+
+
+class PlatformEnv(Env):
+    """
+    Fake poetry Env to match PyPI distributions to the target conda environment
+    """
+
+    def __init__(self, python_version, platform: str):
+        super().__init__(path=Path(sys.prefix))
+        if platform.startswith("linux-"):
+            arch = platform.split("-")[-1]
+            if arch == "64":
+                arch = "x86_64"
+            self._platforms = [
+                f"manylinux{tag}_{arch}" for tag in reversed(MANYLINUX_TAGS)
+            ]
+        elif platform == "osx-64":
+            self._platforms = ["macosx_10_9_x86_64"]
```

Comment:
Does this also need the `universal` platforms as for `arm64`, and maybe also the other `x86_64` wheels?